### DISCUSSION TITLE
Laravel telebot config on database

### DISCUSSION THREAD

**Discussion by ferilagi** [westacks/telebot#99] [Q&A] [answered]
How to save (implement) bot config,
token, name, etc, on database instead on .env, so in production we can easily update value of those setting ?
if any documentation, can anyone share the link or any method to implement my question
thanks in advanced

**Comment by punyflash** [comment]
In case you are using Laravel, I think the simplest way will be adding this logic in your `AppServiceProvider`:

```php
public function boot(): void
{
    Bot::lazy()->each(fn (Bot $bot) => config(["telebot.bots.{$bot->name}" => $bot->createConfig()]));
}
```

Then when your app is bootstrapped you will be able to use all bots using facade

**Reply by ferilagi** [reply]
with that logic, can i use think like this ?
```php
use App\Model\SecretData;
use WeStacks\TeleBot\Laravel\Notifications\TelegramNotification;

 public function toTelegram(object $notifiable): TelegramNotification
    {
    $token = SecretData::where('name', 'token_telegram')->value('data');
    $chatID = $notifiable->telegram_chat_id;

        $invoice = $this->invoice->invoice;
        $name = $this->invoice->name;
        $date = date(now()->toDateString());

        if (isset($chatID)) {
            return (new TelegramNotification( new TeleBot(['token' => $token])))
                ->sendMessage([
                    'chat_id' => $chatID,
                    'text'    => "Hey, this is InvoicePaid Notification ",
                ]);
        } else {
            return 0;
        }
```
 **$token = SecretData::where('name', 'token_telegram')->value('data');**
 i am struggle with using variable $token;

**Reply by punyflash** [reply, marked as answer]
You can, but use `TelegramNotification` correctly:
```php
return (new TelegramNotification)->bot('my-bot')->sendMessage(...)
```

and create the bot in service provicer first:
```php
public function boot(): void
{
    if ($token = SecretData::where('name', 'token_telegram')->value('data')) {
         config(["telebot.bots.my-bot" => compact('token')]);
    }
}
```

**Reply by ferilagi** [reply]
thanks a lot @punyflash, i get it now..
and at last,
all handler must configured in service provider too
or we can build global handler for all bot?

**Reply by punyflash** [reply]
You may create an instance of [Kernel](https://github.com/westacks/telebot/blob/master/src/Kernel.php) class and then use it for every bot:
```php
[
    'token' => $token,
    'handlers' => \App\Telegram\Kernel::class,
]
```
Explained [here](https://westacks.github.io/telebot/#/updates?id=customizing-proccess).

**Reply by ferilagi** [reply]
oke @punyflash, i will try to figure it out
thanks for your fast response
